packages/mui-base/src/ClassNameConfigurator/ClassNameConfigurator.test.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import SwitchUnstyled, { switchUnstyledClasses } from '@mui/base/SwitchUnstyled';
+import ClassNameConfigurator from '@mui/base/ClassNameConfigurator';
+import { createRenderer } from 'test/utils';
+import { expect } from 'chai';
+
+describe('ClassNameConfigurator', () => {
+  const { render } = createRenderer();
+
+  it('should apply default classes when not configured', () => {
+    const { container } = render(<SwitchUnstyled defaultChecked disabled />);
+
+    const switchComponent = container.firstChild!;
+
+    expect(switchComponent).to.have.class(switchUnstyledClasses.root);
+    expect(switchComponent).to.have.class(switchUnstyledClasses.checked);
+    expect(switchComponent).to.have.class(switchUnstyledClasses.disabled);
+
+    expect(switchComponent.childNodes[0]).to.have.class(switchUnstyledClasses.track);
+    expect(switchComponent.childNodes[1]).to.have.class(switchUnstyledClasses.thumb);
+    expect(switchComponent.childNodes[2]).to.have.class(switchUnstyledClasses.input);
+  });
+
+  it('should override the default class names when configured', () => {
+    function classNameGenerator(component: string, slot: string) {
+      return `test:${component.toLowerCase()}/${slot}`;
+    }
+
+    const { container } = render(
+      <ClassNameConfigurator generator={classNameGenerator}>
+        <SwitchUnstyled defaultChecked disabled />
+      </ClassNameConfigurator>,
+    );
+
+    const switchComponent = container.firstChild!;
+
+    expect(switchComponent).to.have.class('test:switch/root');
+    expect(switchComponent).not.to.have.class(switchUnstyledClasses.root);
```

Comment:
Could this turn into quite a challenge for developers that want type safty?

Reply:
If they choose to override the defaults, they can define their own types.